### PR TITLE
Add new initializations

### PR DESCRIPTION
**Pull request by untom**
This PR adds two new initialization methods:

*  The method from "Understanding the difficulty of training deep feedforward neural networks", Glorot & Bengio, AISTATS 2010. This works very well for sigmoid/tanh activation functions. This is known as the "xavier" initialization in [Caffe](http://caffe.berkeleyvision.org/), where it is the default initialization method for dense layers.

* The improvements on the above method made by [He et. al](http://arxiv.org/abs/1502.01852), which they showed works much better when using ReLU activation functions.

### CODE REVIEW

File: keras/initializations.py [hunk 1]
```diff
@@ -20,6 +20,18 @@ def lecun_uniform(shape):
...
+def glorot_normal(shape):
+    ''' Reference: Glorot & Bengio, AISTATS 2010
+    '''
+    s = np.sqrt(2 / (shape[0] + shape[1]))
+    return normal(shape, s)
+
+def he_normal(shape):
+    ''' Reference:  He et al., http://arxiv.org/abs/1502.01852
+    '''
+    s = np.sqrt(2 / (shape[1]))
```
Comment (fchollet) [inline keras/initializations.py]: Integer division would result in 0 in Python 2.7.

**Comment by fchollet** [comment]
A few things:

- Use float divisions for Python 2.7 compatibility
- Activations can potentially be used anywhere, on any input shape, so should not be dependent on the number of dimensions of the input.
- I believe the scheme proposed by He is sqrt(2/n_l) where n_l is the number of elements in the layer (n_l = k*k*c, with k*k the dimensions of the 2D input and c the number of channels).

**Comment by untom** [comment]
Silly me for forgetting to include ``from __future__ import division``. You're right, I didn't consider convolutional layers. Should I fix those issues and resubmit or is this something you're not interested in anyhow?

**Comment by fchollet** [comment]
I think these initializations (and in general, more initializations) are definitely valuable to have, of course! If you can generalize them to be shape-agnostic and use float division, that'd be great : )
